### PR TITLE
Introduce a list command to list all transients

### LINKED ISSUE #44: Add list command for listing transients stored in the database
For debugging transients stored in the database a `wp transient list` command would be helpful. Related plugins are [Debug Bar Transients ](https://wordpress.org/plugins/debug-bar-transients/) or [Transients Manager](https://wordpress.org/plugins/transients-manager/).

### PR DESCRIPTION
**Pull request by ocean90**
Fixes #44.

### CODE REVIEW

File: src/Transient_Command.php [hunk 1]
```diff
@@ -220,6 +220,139 @@ public function type() {
...
 	}
 
+	/**
+	 * Lists transients and their values.
+	 *
+	 * ## OPTIONS
+	 *
+	 * [--network]
+	 * : Get the values of network|site transients. On single site, this is
+	 * is a specially-named cache key. On multisite, this is a global cache
```
Comment (schlessera) [inline src/Transient_Command.php]:
```suggestion
	 * a specially-named cache key. On multisite, this is a global cache
```

File: features/transient.feature [hunk 2]
```diff
@@ -27,6 +27,29 @@ Feature: Manage WordPress transient cache
...
+    And I run `wp transient set foo bar`
+    And I run `wp transient set foo2 bar2 610`
+    And I run `wp transient set foo3 bar3 --network`
+    And I run `wp transient set foo4 bar4 610 --network`
+
+    When I run `wp transient list --format=csv`
+    Then STDOUT should be:
+      """
+      name,value,expiration
+      foo,bar,"no expiration"
```
Comment (schlessera) [inline features/transient.feature]: On the non-human output, should the expiration here return `false`?

File: src/Transient_Command.php [hunk 3]
```diff
@@ -220,6 +220,139 @@ public function type() {
...
+			return 'no expiration';
+		}
+
+		$now = time();
+
+		if ( $now > $expiration ) {
+			return 'expired';
+		}
+
+		return human_time_diff( $now, $expiration );
```
Comment (ocean90) [inline src/Transient_Command.php:435]: Make this machine readable by default

File: src/Transient_Command.php [hunk 4]
```diff
@@ -220,6 +220,139 @@ public function type() {
...
+	 * ---
+	 * default: table
+	 * options:
+	 *   - table
+	 *   - json
+	 *   - csv
+	 *   - count
+	 *   - yaml
+	 * ---
+	 *
```
Comment (ocean90) [inline src/Transient_Command.php:271]: Let's also add arguments for filtering.

**Review by schlessera — CHANGES_REQUESTED** [review]

**Review by schlessera — APPROVED** [review]

**Comment by schlessera** [comment]
Thanks for the PR, @ocean90 !

**Pull request merged**